### PR TITLE
Fix SDL sends incorrect result on duplicated menuName

### PR DESCRIPTION
**Pull request by ATymchenko**
Added check for duplicated menuName

### CODE REVIEW

**Comment by indlu** [comment]
@AByzhynar, @okozlovlux, @LevchenkoS , please review

File: src/components/application_manager/src/commands/mobile/create_interaction_choice_set_request.cc [hunk 1]
```diff
@@ -151,6 +151,10 @@ mobile_apis::Result::eType CreateInteractionChoiceSetRequest::CheckChoiceSet(
     for (next_choice_set_it = current_choice_set_it + 1;
          choice_set->end() != next_choice_set_it;
          ++next_choice_set_it) {
+      if (compareStr((*current_choice_set_it)[strings::menu_name],
+                     (*next_choice_set_it)[strings::menu_name])) {
+        return mobile_api::Result::DUPLICATE_NAME;
+      }
       if (compareSynonyms(*current_choice_set_it, *next_choice_set_it)) {
```
Comment (okozlovlux) [inline src/components/application_manager/src/commands/mobile/create_interaction_choice_set_request.cc]: @indlu , think this `if(..` is redundant

Comment (indlu) [inline src/components/application_manager/src/commands/mobile/create_interaction_choice_set_request.cc]: @okozlovlux, fixed in 794d783

File: src/components/application_manager/src/commands/mobile/create_interaction_choice_set_request.cc [hunk 2]
```diff
@@ -151,8 +151,13 @@ mobile_apis::Result::eType CreateInteractionChoiceSetRequest::CheckChoiceSet(
     for (next_choice_set_it = current_choice_set_it + 1;
          choice_set->end() != next_choice_set_it;
          ++next_choice_set_it) {
-      if (compareSynonyms(*current_choice_set_it, *next_choice_set_it)) {
-        return mobile_apis::Result::DUPLICATE_NAME;
+      const bool menu_name_duplicated =
+          compareStr((*current_choice_set_it)[strings::menu_name],
+                     (*next_choice_set_it)[strings::menu_name]);
+      const bool choice_set_duplicated =
+          compareSynonyms(*current_choice_set_it, *next_choice_set_it);
```
Comment (okozlovlux) [inline src/components/application_manager/src/commands/mobile/create_interaction_choice_set_request.cc]: @indlu , why do you compare `*current_choice_set_it, *next_choice_set_it` if you just need to compare `menu_name` ?

Comment (indlu) [inline src/components/application_manager/src/commands/mobile/create_interaction_choice_set_request.cc]: @okozlovlux, current_choice_set must be different, menu_name also must be different.
In the both cases we return mobile_api::Result::DUPLICATE_NAME.

**Comment by AByzhynar** [comment]
@indlu Add @LuxoftAKutsan or @dev-gh to code review please

**Comment by AByzhynar** [comment]
@indlu Remove @LevchenkoS as he is not working for our project

**Comment by jacobkeeler** [comment]
No longer needed after #2346

**Pull request closed without merging**